templates/terraform/constants/health_check.erb

```diff
@@ -0,0 +1,47 @@
+// Whether the port should be set or not
+// TODO: With the next breaking change we should remove the defaultPort.
+// We've added the ability to check if port is set to the defaultPort here,
+// then just remove it in the encoder.  However, a better fix would be to
+// not set the default port, but this will be a breaking change.
```

Comment:
Is it a breaking change to not have the default in the schema but to set the port value in the encoder to the expected default? I think that would probably be my approach - we'd have computed port. If a user tries to set port with a non-port port specification, they can just have an error IMO, so I think if we avoid using the schema default port to actually set default, we can avoid the permadiff.

Reply:
I like this idea, and will try it out.  Thanks!